code/__HELPERS/_lists.dm

```diff
@@ -27,6 +27,8 @@
 #define LAZYACCESS(L, I) (L ? (isnum(I) ? (I > 0 && I <= length(L) ? L[I] : null) : L[I]) : null)
 ///Sets the item K to the value V, if the list is null it will initialize it
 #define LAZYSET(L, K, V) if(!L) { L = list(); } L[K] = V;
+///Sets the length of a lazylist
+#define LAZYSETLEN(L, V) if (!L) { L = list(); } L.len = V;
```

Comment:
This is a *really* niche necessity, what do you need this for?

Reply:
To avoid repeating LAZYINITLIST calls followed by setting the list length the next line. After all, lazy list macros exist to avoid having to repeat certains operations.
And If you are wondering, trying to use LAZYSET with a numerical index will throw list out of bounds expections if the index is higher than its length, so setting the length beforehand is necessary.